src/node.cpp

```diff
@@ -329,8 +328,15 @@ Node* Node::unserialize(MegaClient* client, const string* d, node_vector* dp)
     hasLinkCreationTs = MemAccess::get<char>(ptr);
     ptr += sizeof(hasLinkCreationTs);
 
-    hasLinkAuthKey = MemAccess::get<char>(ptr);
-    ptr += sizeof(hasLinkAuthKey);
+    auto authKeySize = MemAccess::get<char>(ptr);
```

Comment:
Please, don't use `auto` here. It's important that the datatype is `char` and not any other.